kolibri/core/device/models.py

```diff
@@ -210,9 +213,34 @@ def clean_stale(cls, expire=180.0):
         cls.objects.filter(updated__lte=staled_time).delete()
 
 
+class SyncStatusPermissions(DenyAll):
+    """
+    By default set all permissions to read only, as these should
+    only be directly updated by internal Kolibri processes, not by users.
+    """
+
+    def user_can_read_object(self, user, obj):
+        if hasattr(user, "dataset"):
+            return obj.user.dataset_id == user.dataset_id
+        return False
+
+    def readable_by_user_filter(self, user):
+        """
+        For simplicity, let anyone from the same facility read the sync status
+        for now. In the future, we may want to restrict this, but it is not
+        sensitive information.
```

Comment:
The model contains a foreignkey onto SyncSession, which means it directly stores the sync session ID, which is used as the "session token" (like a browser session_id cookie) for purposes of syncing -- if the syncsession is still active, anybody with that ID can sync within the scope of that sync session, without further certificate verification.

I'm not seeing anywhere in the codebase yet that pulls this model in to display in the frontend, but I'd recommend that it not send the sync session ID as part of that, instead just looking up the relevant fields (e.g. last_activity timestamp) across the foreignkey. If that were enforced, then it would perhaps be OK to leave these permissions as-is. But the safest would be to only allow access for someone who's a coach or admin for the target user.